authorization/v2/uniformResourceIdentifier.go

```diff
@@ -52,3 +127,31 @@ func getRequestArguments(ctx *gin.Context) map[string]string {
 
 	return args
 }
+
+func unmarshallURIList(source string) (map[string]string, error) {
+	uriListMapping := map[string]string{}
+	keyValuePairs := strings.Split(source, "&")
+
+	for index, keyValuePair := range keyValuePairs {
+		split := strings.Split(keyValuePair, "=")
+		if len(split) != 2 {
+			return nil, errors.New("malformed key value pair at index " + string(index))
+		}
+		uriListMapping[split[0]] = split[1]
+	}
+
+	return uriListMapping, nil
+}
+
+func marshallURIMap(uriMap map[string]string) string {
+	var marshalledMap string
+	if uriMap == nil || len(uriMap) == 0 {
+		return marshalledMap
+	}
+
+	for key, value := range uriMap {
+		marshalledMap += key + "=" + value + "&"
+	}
+
+	return marshalledMap[:len(marshalledMap)-1]
```

Comment:
Could you add a little comment about why you're removing the last character just for readability's sake? It's not immediately obvious that you're removing the extra `&`.